internal/pkg/agent/application/coordinator/coordinator_test.go

```diff
@@ -313,6 +314,78 @@ func TestCoordinator_StateSubscribe(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestCoordinatorWithErrors(t *testing.T) {
+	handlerChan, runtime, varWatcher, config := setupAndWaitCoordinatorDone()
+
+	close(runtime)
+	close(varWatcher)
+	cfgErrStr := "configWatcher error"
+	config <- errors.New(cfgErrStr)
+
+	// All the multierror stuff breaks errors.Is
+	waitAndTestError(t, func(err error) bool { return strings.Contains(err.Error(), cfgErrStr) }, handlerChan)
+
+}
+
+func TestCoordinatorShutdownClosedChannels(t *testing.T) {
+
+	handlerChan, runtime, varWatcher, config := setupAndWaitCoordinatorDone()
+
+	close(runtime)
+	close(varWatcher)
+	close(config)
+
+	waitAndTestError(t, func(err error) bool { return errors.Is(err, context.Canceled) }, handlerChan)
+
+}
+
+func TestCoordinatorShutdownTimeout(t *testing.T) {
+	CoordinatorShutdownTimeout = time.Millisecond
+	handlerChan, _, _, _ := setupAndWaitCoordinatorDone()
+	waitAndTestError(t, func(err error) bool { return errors.Is(err, context.Canceled) }, handlerChan)
```

Comment:
I would also have a test where one channel returns an error, one channel closes, and one channel never does anything to test the combination of all three event types together.